internal/resources/cluster.go

```diff
@@ -423,6 +433,16 @@ func (c *Cluster) Update(ctx context.Context, req resource.UpdateRequest, resp *
 		if clusterapi.AreEqual(plan.ServiceGroups[i].Services, serviceGroup.Services) {
 			currentState.ServiceGroups[i].Services = plan.ServiceGroups[i].Services
 		}
+
+		// for Azure clusters - Premium disk types, if the user provides the storage and iops values (these values are ignored), then put the value provided in state file,
+		// as otherwise there will be a type mismatch with the actual value - i.e. the default values for premium disks.
+		if currentState.CloudProvider.Type == types.StringValue("Azure") {
```

Comment:
Azure should be a const, as I assume it will get used at multiple places.
same for other strings below.